### ISSUE TITLE
puppet-lint --fix encountered an error that it doesn't know how to handle

### ISSUE THREAD

**Issue by kalvaer** [rodjek/puppet-lint#706]
---
puppet-lint version: 2.2.1
ruby version: 2.0.0-p648
platform: universal.x86_64-darwin16
file path: ./code/manifests/trig_conf.pp
file contents:
```
/*****************************************************************************
 * Puppet template for the atlas-trig-conf webservice
 *
 *  William Strecker-Kellogg <willsk@bnl.gov>
 *
 ****************************************************************************/

class hg_voatlasmisc::trig_conf {

    include cvmfs
    $app_docroot = '/var/www/trigconf'

    class { 'hg_voatlasmisc::servicebase': }

    class { 'hg_voatlasmisc::servicebase::gitdeploy':
        reponame    => 'trigconf',
        deploydir   => $app_docroot,
    }

    file { '/etc/sysconfig/httpd':
        ensure  => present,
        source  => 'puppet:///modules/hg_voatlasmisc/trig_conf/httpd-sysconfig.conf',
        owner   => 'root',
        group   => 'root',
        mode    => '0644',
    }

    class { 'apache::mod::wsgi':
        wsgi_socket_prefix  => '/var/run/wsgi',
    }

    file { ['/data/', '/data/bunchgroups','/data/bunchgroups/plots', '/data/cron']:
        ensure  => directory,
        owner   => 'apache',
    }


    $present_in_prod = $::web_instance_type ? {
        prod    => 'present',
        default => 'absent',
    }

    ## just for devel and test instance if needed
    #if $present_in_prod == 'absent' {
        file { '/etc/httpd/conf.d/00_wsgi.load':
          ensure => 'present',
          replace => true,
          owner  => 'root',
          group  => 'root',
          mode   => '0644',
          content => "
# Loading custom module
LoadModule wsgi_module /var/www/trigconf/wsgi-current/mod_wsgi.so
",
          notify => Service["httpd"],
      }
    #}

    hg_voatlasmisc::servicebase::mount_point { 'trig-conf-data':
        ensure  => $present_in_prod,
        path    => '/data/cron',
        require => File['/data/cron'],
    }

    cron { 'trigconf-wtrp-dump':
        ensure  => $present_in_prod,
        command => '/var/www/trigconf/bootstrap/wtrp_dump.sh',
        user    => 'root',
        hour    => '23',
        minute  => '59',
    }

    /* This stuff can't use the built-in upstream apache wsgi_* settings because
     * the implementation of that is broken -- the order of script-aliases
     * changes randomly with hash-traversal in ruby, which is very stupid
     * because order matters here.
     * Also, the manifest doesn't support multiple WSGIDaemonProcess's, which
     * is needed here.
     * Will - Sept. 2014
     */

    $wsgi_stuff = "
  WSGIDaemonProcess trigconf-bunchgroups display-name=trigconf-bunchgroups group=zp processes=5 threads=1
  WSGIScriptAlias /bunchgroups \"/var/www/trigconf/apps/bunchgroups/django.wsgi\" process-group=trigconf-bunchgroups

  WSGIDaemonProcess trigconf-timelines display-name=trigconf-timelines group=zp processes=5 threads=10
  WSGIScriptAlias /timelines \"/var/www/trigconf/apps/timelines/django.wsgi\" process-group=trigconf-timelines

  WSGIDaemonProcess trigconf-main display-name=trigconf-main group=zp processes=5 threads=10
  WSGIScriptAlias / \"/var/www/trigconf/apps/trigconf/django.wsgi\" process-group=trigconf-main

  # For large headers
  LimitRequestFieldSize 262144
"


    apache::vhost { 'trigconf-vhost':
        servername          => 'default',
        docroot             => $app_docroot,
        docroot_owner       => 'apache',
        port                => 443,
        ssl                 => true,
        aliases             => [
            {
                alias => '/bunchgroups/graphics',
                path  => '/data/bunchgroups',
            }, {
                alias => '/media/',
                path  => "${app_docroot}/media/",

            }, {
                alias => '/wtrp_archive/',
                path  => '/data/cron/wtrp_archive/',
            }, {
                alias => '/ct_cool/',
                path  => "${app_docroot}/ct_cool",
            }, {
                alias => '/l1ct/',
                path  => "${app_docroot}/l1ct",
            }, {
                alias => '/timelines/favicon.ico',
                path  => "${app_docroot}/media/images/favicon.ico",
            }
        ],

        directories         => [
            {
                path            => "${app_docroot}/apps/trigconf/run/",
                custom_fragment => 'WSGIApplicationGroup trigconf-menus',
            }, {
                path            => "${app_docroot}/apps/",
                custom_fragment => 'WSGIApplicationGroup %{GLOBAL}',
            }, {
                path            => "${app_docroot}/apps/bunchgroups",
                custom_fragment => 'WSGIApplicationGroup %{GLOBAL}',
            }, {
                path            => "${app_docroot}/apps/timelines",
                custom_fragment => 'WSGIApplicationGroup trigconf-timelines',
            }

        ],
        options             => ['Indexes', 'FollowSymLinks', ],

        custom_fragment     => $wsgi_stuff,
    }

##  class { 'afs':
##    memcachesize => "4194304",
##    options   => {
##      "memcache" => true,
##    },
##    cachesize => 'AUTOMATIC'
##  }
    
}

```
error:
```
NoMethodError: undefined method `next_token=' for nil:NilClass
/Library/Ruby/Gems/2.0.0/gems/puppet-lint-2.2.1/lib/puppet-lint/plugins/check_whitespace.rb:49:in `fix'
/Library/Ruby/Gems/2.0.0/gems/puppet-lint-2.2.1/lib/puppet-lint/checkplugin.rb:41:in `block in fix_problems'
/Library/Ruby/Gems/2.0.0/gems/puppet-lint-2.2.1/lib/puppet-lint/checkplugin.rb:38:in `each'
/Library/Ruby/Gems/2.0.0/gems/puppet-lint-2.2.1/lib/puppet-lint/checkplugin.rb:38:in `fix_problems'
/Library/Ruby/Gems/2.0.0/gems/puppet-lint-2.2.1/lib/puppet-lint/checks.rb:61:in `block in run'
/Library/Ruby/Gems/2.0.0/gems/puppet-lint-2.2.1/lib/puppet-lint/checks.rb:56:in `each'
/Library/Ruby/Gems/2.0.0/gems/puppet-lint-2.2.1/lib/puppet-lint/checks.rb:56:in `run'
/Library/Ruby/Gems/2.0.0/gems/puppet-lint-2.2.1/lib/puppet-lint.rb:191:in `run'
/Library/Ruby/Gems/2.0.0/gems/puppet-lint-2.2.1/lib/puppet-lint/bin.rb:59:in `block in run'
/Library/Ruby/Gems/2.0.0/gems/puppet-lint-2.2.1/lib/puppet-lint/bin.rb:56:in `each'
/Library/Ruby/Gems/2.0.0/gems/puppet-lint-2.2.1/lib/puppet-lint/bin.rb:56:in `run'
/Library/Ruby/Gems/2.0.0/gems/puppet-lint-2.2.1/bin/puppet-lint:7:in `<top (required)>'
/usr/local/bin/puppet-lint:23:in `load'
/usr/local/bin/puppet-lint:23:in `<main>'
```

**Comment by kalvaer** [comment]
Issue is due to the comments with /*
Actually looking at that code, I cant see how it could be fixed by puppet lint at all ;)
Ignoring the check and everything else worked

**Comment by rodjek** [comment]
Reopening as this should still be handled in a nicer way than a stacktrace :)

**Comment by rodjek** [comment]
@kalvaer I can't reproduce this locally, could you send me a link to the actual file as there might be a unicode character or something in it that isn't being represented in the paste?

**Comment by kalvaer** [comment]
[trig_conf copy.txt](https://github.com/rodjek/puppet-lint/files/1248056/trig_conf.copy.txt)
I have since changed the file to remove the /* comments.
This should be it though from previous commits

**Comment by rodjek** [comment]
Thanks, but this looks like a different commit than the pasted content going by the header comment. I'm unable to reproduce the problem with this file either. Is there an older commit for the file that I can try?

**Comment by kalvaer** [comment]
[trig_conf.pp.txt](https://github.com/rodjek/puppet-lint/files/1248618/trig_conf.pp.txt)

This was from 2 years ago when the comments where added by the author

**Comment by rodjek** [comment]
Thanks, I've reproduced the problem, will have a patch to fix it up shortly :)

**Comment by rodjek** [comment]
This has been fixed up as part of #694
